src/pubsub/mod_pubsub_db_rdbms_sql.erl

```diff
@@ -263,16 +267,195 @@ update_subscription(Nidx, LU, LS, LR, SubInt, SubId) ->
 
 % ------------------- Items --------------------------------
 
--spec insert_item(Nidx :: mod_pubsub:nodeIdx(),
-                  LU :: jid:luser(),
-                  LS :: jid:lserver(),
-                  ItemId :: mod_pubsub:itemId()) -> iolist().
-insert_item(Nidx, LU, LS, ItemId) ->
-    ["INSERT INTO pubsub_items (nidx, luser, lserver, itemid)"
-     " VALUES (", esc_int(Nidx), ", ",
-     esc_string(LU), ", ",
-     esc_string(LS), ", ",
-     esc_string(ItemId), ")"].
+-spec get_items(Nidx :: mod_pubsub:nodeIdx(), gen_pubsub_node:get_item_options()) ->
+    iolist().
+get_items(Nidx, Opts) ->
+    ["SELECT ", item_columns(), " ",
+     "FROM pubsub_items "
+     "WHERE nidx=", esc_int(Nidx),
+     maybe_item_ids_filter(maps:get(item_ids, Opts, undefined)),
+     " ORDER BY modified_at DESC",
+     maybe_result_limit(maps:get(max_itemx, Opts, undefined))].
+
+-spec get_item(mod_pubsub:nodeIdx(), mod_pubsub:itemId()) -> iolist().
+get_item(Nidx, ItemId) ->
+    ["SELECT ", item_columns(), " ",
```

Comment:
How about adding `" "` to next item in list for better performance:
```sql 
["SELECT ", item_columns(), 
" FROM pubsub_items ",
"WHERE nidx=", esc_int(Nidx),
" AND itemid=", esc_string(ItemId)]
```
??
Less elements in a list -> quicker concatenation

Reply:
I don't like the proposed formatting :) to eliminate the one element from list I'll do the following:

```erlang
    ["SELECT ", item_columns(), " "
     "FROM pubsub_items "
     "WHERE nidx=", esc_int(Nidx),
     " AND itemid=", esc_string(ItemId)].
```

mind the missing `,` after strings. Thanks to that the compiler concatenates these to on string `" FROM pubsub_items WHERE nidx="`. I think this is even more optimized than what you suggested :) 🥇